### ISSUE TITLE
GML export/import breaks on bools

### ISSUE THREAD

**Issue by domoritz** [networkx/networkx#2118]
``` python
import networkx as nx

G = nx.DiGraph()
G.add_node(1)
G.node[1]['foo'] = True
nx.write_gml(G, 'foo.gml')

H = nx.read_gml('foo.gml')
```

Crashes with

```
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    H = nx.read_gml('foo.gml')
  File "<decorator-gen-198>", line 2, in read_gml
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/utils/decorators.py", line 220, in _open_file
    result = func(*new_args, **kwargs)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 210, in read_gml
    G = parse_gml_lines(filter_lines(path), label, destringizer)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 383, in parse_gml_lines
    graph = parse_graph()
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 372, in parse_graph
    curr_token, dct = parse_kv(next(tokens))
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 357, in parse_kv
    curr_token, value = parse_dict(curr_token)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 367, in parse_dict
    curr_token, dct = parse_kv(curr_token)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 357, in parse_kv
    curr_token, value = parse_dict(curr_token)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 367, in parse_dict
    curr_token, dct = parse_kv(curr_token)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 343, in parse_kv
    curr_token = next(tokens)
  File "/Users/domoritz/.virtualenvs/ipython/lib/python2.7/site-packages/networkx/readwrite/gml.py", line 323, in tokenize
    (line[pos:], lineno + 1, pos + 1))
networkx.exception.NetworkXError: cannot tokenize u'True' at (6, 9)
```

**Comment by chebee7i** [comment]
Looks like a bug here: https://github.com/networkx/networkx/blob/master/networkx/readwrite/gml.py#L587

```
>>> isinstance(True, bool), str(True)
True, 'True'
```

**Comment by hagberg** [comment]
Looks like GML bools are 0,1 so we should fix this bug to parse and generate those correctly.

**Comment by chebee7i** [comment]
My comment was utterly unhelpful. Note to self: Do not link to a line in master, only link to a line on a fixed commit.

The line I intended to highlight was:

```
if isinstance(value, (int, long)):
```

In this case, the boolean passes through and is converted to a string.

**Comment by dschult** [comment]
Fixed by #2513
